### PR TITLE
Add analytics to `mlstacks`

### PR DESCRIPTION
**Pull request by strickvl**
Adds simple analytics to use of CLI. Also added documentation for analytics + for opt-out.

### CODE REVIEW

File: src/mlstacks/utils/analytics_utils.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Analytics utils for MLStacks."""
+
+import sys
+
+
+def python_version() -> str:
```
Comment (bcdurak) [inline src/mlstacks/utils/analytics_utils.py:18]: In order to get this without parsing it yourself and keep it consistent with the main repo you can use:

```python
import platform
p_version =  platform.python_version()
```

File: src/mlstacks/cli/cli.py
Comment (bcdurak) [inline src/mlstacks/cli/cli.py:1]: For the CLI functionality that you would like to track, we can look into [the track decorator](https://github.com/zenml-io/zenml/blob/8951f3b103282127eef3de9186f70585c9490663/src/zenml/utils/analytics_utils.py#L570), which inherently uses the `EventHandler` and lets you avoid the indentation.

**Review by bcdurak — CHANGES_REQUESTED** [review]
Overall, everything looks good. This is a great start for enabling analytics for MLStacks. I left a few small comments. Feel free to ignore any of them.

File: pyproject.toml [hunk 2]
```diff
@@ -74,6 +74,7 @@ pyyaml = { version = "6.0" }
 click = { version = "8.1.3" }
 python-terraform = { version = "^0.10.1" }
 rich = { version = "^13.5.2" }
+segment-analytics-python = { version = "^2.2.3" }
```
[the remark this answers is not in the document]
Comment (strickvl) [reply pyproject.toml:77]: Nice catch. I'll do that.

File: src/mlstacks/analytics/client.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+            exc_type (Optional[Type[BaseException]]): Exception type
+            exc_val (Optional[BaseException]): Exception value
+            exc_tb (Optional[TracebackType]): Traceback type
+
+        Returns:
+            True if no exception occurred, False otherwise
+        """
+        if exc_val:
+            # Handle exception logging if necessary. Here I'm just printing it.
+            print(f"Error occurred: {exc_val}")  # noqa: T201
```
[the remark this answers is not in the document]
Comment (strickvl) [reply src/mlstacks/analytics/client.py:84]: Will update. I think this was just a copy-paste from the zenml repo, but your suggestion makes sense.

File: src/mlstacks/cli/cli.py
Comment (strickvl) [reply src/mlstacks/cli/cli.py:1]: Will make a note to tackle that later, but for now I think probably will leave it like it is.

**Review by bcdurak — APPROVED** [review]
🦭

**Pull request merged**